pkg/sql/ttl/ttljob/ttljob_test.go

```diff
@@ -299,17 +337,69 @@ INSERT INTO t (id, crdb_internal_expiration) VALUES (1, now() - '1 month'), (2,
 			th.sqlDB.ExecMultiple(t, strings.Split(tc.setup, ";")...)
 
 			// Force the schedule to execute.
-			th.env.SetTime(timeutil.Now().Add(time.Hour * 24))
-			require.NoError(t, th.executeSchedules())
-
 			th.waitForScheduledJob(t, jobs.StatusFailed, tc.expectedTTLError)
+
 			var numRows int
 			th.sqlDB.QueryRow(t, `SELECT count(1) FROM t`).Scan(&numRows)
 			require.Equal(t, 2, numRows)
 		})
 	}
 }
 
+func TestRowLevelTTLJobMultipleNodes(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	th, cleanupFunc := newRowLevelTTLTestJobTestHelper(
+		t,
+		&sql.TTLTestingKnobs{
+			AOSTDuration:                  &zeroDuration,
+			ReturnStatsError:              true,
+			RequireMultipleSpanPartitions: true,
+		},
+		false, /* testMultiTenant */ // SPLIT AT does not work with multi-tenant
+		5,     /* numNodes */
+	)
+	defer cleanupFunc()
+
+	sqlDB := th.sqlDB
+
+	sqlDB.Exec(t, `CREATE TABLE tbl (
+			id INT PRIMARY KEY,
+			expire_at TIMESTAMP
+		) WITH (ttl_expiration_expression = 'expire_at')`,
+	)
+
+	splitAt := 10_000
+	sqlDB.Exec(t, `ALTER TABLE tbl SPLIT AT VALUES ($1)`, splitAt)
+
+	expectedNumNonExpiredRows := 0
+	expectedNumExpiredRows := 0
+	ts := timeutil.Now()
+	nonExpiredTs := ts.Add(time.Hour * 24 * 30)
+	expiredTs := ts.Add(-time.Hour)
+	// even pk is non-expired, odd pk is expired
+	for i := 0; ; i += 2 {
+		sqlDB.Exec(t, `INSERT INTO tbl VALUES ($1, $2)`, i, nonExpiredTs)
+		sqlDB.Exec(t, `INSERT INTO tbl VALUES ($1, $2)`, splitAt+i, nonExpiredTs)
+		expectedNumNonExpiredRows += 2
+		sqlDB.Exec(t, `INSERT INTO tbl VALUES ($1, $2)`, i+1, expiredTs)
+		sqlDB.Exec(t, `INSERT INTO tbl VALUES ($1, $2)`, splitAt+i+1, expiredTs)
+		expectedNumExpiredRows += 2
+		ranges := sqlDB.QueryStr(t, `SHOW RANGES FROM TABLE tbl`)
+		if ranges[0][4] != ranges[1][4] {
```

Comment:
This seems kind of hacky.  Is there a better way to ensure multiple leaseholders?